pkg/client/s3client.go

```diff
@@ -39,12 +40,50 @@ type S3Client struct {
 	StorageClass string
 	SvcEndpoint  string
 	S3Session    *s3.S3
+	AccessKey    string
+	SecretKey    string
 }
 
 const (
 	AuthEndpoint = "https://iam.cloud.ibm.com/identity/token"
 )
 
+//Func NewS3Client accepts apikey, accesskey, secretkey of the bucket and return the s3 client
+//to perform different s3 operations like upload, delete etc.,
+func NewS3Clientwithkeys(c *Client, region string) (s3client *S3Client, err error) {
+
+	s3client = &S3Client{}
+	if pkg.Options.APIKey == "" {
+		s3client.ApiKey = os.Getenv("IBMCLOUD_API_KEY")
+	} else {
+		s3client.ApiKey = pkg.Options.APIKey
+	}
+
+	if pkg.ImageCMDOptions.AccessKey != "" && pkg.ImageCMDOptions.SecretKey != "" {
+		s3client.AccessKey = pkg.ImageCMDOptions.AccessKey
+		s3client.SecretKey = pkg.ImageCMDOptions.SecretKey
```

Comment:
Lets not access these variables directly, lets pass them to the `NewS3Clientwithkeys` function to avoid direct deps on the pkg.ImageCMDOptions

Reply:
Done.